sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -87,6 +89,8 @@ public virtual async Task ReceiveAndProcessMessagesAsync(CancellationToken cance
                     {
                         continue;
                     }
+
+                    using var bodyScope = message.CreateBodyScope();
```

Comment:
I appreciate what you are trying to do here, but what if a user holds on to the message reference after their callback returns? The body would be gone.

Reply:
Hence my original proposal in https://github.com/Azure/azure-sdk-for-net/pull/19885 to make it opt-in. This can still be done by exposing this opt in via the processor options no? And if you do this you could even make it opt-in in a minor and then obsolete with warn and make it opt-out in future major versions if you worry about existing code that might do something like that.

Reply:
Yeah, if we did this it would definitely need to be opt-in. We don't obsolete with warn as that could be treated as an error in different scenarios.

Can you clarify what you mean when you say it is now stable memory? It looks like it is still increasing linearly, but just a lot less memory used.

Reply:
The linear memory increase you see on the screenshots is already there and comes from the AMQP library mostly. The Guid conversion allocations are from the code pointed out in https://github.com/Azure/azure-sdk-for-net/issues/20166

By stable I meant the fact that there are no Gen0 allocations anymore for the body for messages received by the processor. So in essence the memory is now predictably stable for the bodies based on the maximum expected message size as well as the concurrency.

Reply:
When you say there are no Gen0 allocations anymore, can you point to that in the screenshot?

Reply:
All the screenshots show the memory traffic of `byte[]` during the receive of 10k or 20k messages with the following content

```
var message = new ServiceBusMessage(UTF8.GetBytes($"Deep Dive {i} Deep Dive {i} Deep Dive {i} Deep Dive {i} Deep Dive {i} Deep Dive {i}"));
```

In the after screenshot you can see that the `Array.Resize` traffic is gone.